libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -495,6 +495,25 @@ void RC_Channel::init_aux_function(const aux_func_t ch_option, const aux_switch_
     }
 }
 
+#if !HAL_MINIMIZE_FEATURES
+
+/* lookup the announcement for switch change*/
+const char *RC_Channel::string_for_aux_function(AUX_FUNC function) const     
+{
+    for (int k=0; 48; k++) {     //wish I knew how to determine the # rows in the table!
```

Comment:
`ARRAY_SIZE(lookuptable)`

Alternatively you can just end the table with a sentinel and detect that (IE a `nullptr` for the string means you are done iterating the table)

Reply:
@WickedShell  I think I am doing that now.,,I originally had the test at 255 ...but was afraid that if someone accidentally deleted the end marker that I would march off into the badlands and bring back something bad
...but can I declare the table in the .h with no size ahead of time...it  LookupTable lookuptable[] {   instead of a fixed size? 
if so, that would make additions a lot simpler

Reply:
The for loop structure here isn't actually helping, since you aren't using it as a for loop. You might as well use the ARRAY_SIZE macro, or use the for each as suggested below and then you can save the time/comparison for the sentinel, and you avoid the accidental wandering off the end.

Reply:
thanks, done!